packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -537,7 +553,17 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       _pendingDimensions = false;
     }
     assert(!_didChangeViewportDimensionOrReceiveCorrection, 'Use correctForNewDimensions() (and return true) to change the scroll offset during applyContentDimensions().');
-    _lastMetrics = copyWith();
+
+    if (_isMetricsChanged()) {
+      // It isn't safe to trigger the ScrollMetricsNotification if we are in
+      // the middle of rendering the frame, the developer is likely to schedule
+      // a new frame(build scheduled during frame is illegal).
+      if (_lastMetrics != null && !_haveScheduledUpdateNotification) {
```

Comment:
Adjusted the notification sending strategy:
1. Consumers are not notified when `_lastMetrics` is initialized for the first time.
2. Merge multiple change events in the same frame, and only notify the last result.